core/assertion.ts

```diff
@@ -213,6 +213,12 @@ export class AssertionContext implements ICommonContext {
     );
   }
 
+  public database(): string {
+    return this.assertion.session.finalizeDatabase(
+      this.assertion.proto.target.database
```

Comment:
we may be operating in a `database`-less warehouse, e.g. redshift/postgres. this method should probably throw `if (!this.assertion.proto.target.database)`

Reply:
Good points, have updated.